src/Storage/Storage/Blob/Cmdlet/GetAzureStorageBlobContent.cs

```diff
@@ -280,12 +282,33 @@ internal string GetFullReceiveFilePath(string fileName, string blobName, DateTim
             return filePath;
         }
 
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                FileName = GetUnresolvedProviderPathFromPSPath(FileName);
+                Validate.ValidateInternetConnection();
+                InitChannelCurrentSubscription();
+                this.ExecuteSynchronouslyOrAsJob();
+            }
+            catch (Exception ex) when (!IsTerminatingError(ex))
+            {
+                WriteExceptionError(ex);
+            }
+        }
+
+
```

Comment:
Following code should be added here, or the DoBeginProcessing() will be run 2 times when use -asjob parameter. (once in BeginProcessing(), once in  ExecuteCmdlet())

I have tested before/after adding the code, the cmdlets still runs successfully with/without -asjob.

```
        protected override void BeginProcessing()
        {
            if (!AsJob.IsPresent)
            {
                DoBeginProcessing();
            }
        }
```

Reply:
this is intentional - DoBeginProcessing is idempotent in this case